### ISSUE TITLE
Crashing on firefox and safari

### ISSUE THREAD

**Issue by danillocesar** [greatCodeIdeas/md-date-range-picker#27]
Just open the demo page on those browser and you will see..

Any quick fix for that?

**Comment by roelbarreto** [comment]
@danillocesar  Works fine on my firefox Windows, I have no mac yet to test it but I'll find a way to test on Mac Safari/Firefox
![image](https://user-images.githubusercontent.com/1954163/30630352-d6c490b2-9e12-11e7-94d4-74ccf22f60e6.png)

**Comment by danillocesar** [comment]
That's weird. Here's what happens on my firefox and on my clients.
![ff](https://user-images.githubusercontent.com/20955833/30647147-4432855e-9df1-11e7-87e3-0f8b565c9f22.png)

It takes a lot of time to load then firefox ask me to shut it down because some script is taking too long to run, probably in some kinda of a infinite loop.

**Comment by roelbarreto** [comment]
I'll Investigate it

**Comment by danillocesar** [comment]
I'm in Brazil, if that helps. From what i've seen, its on a method that change the time of the calendar.. its possiblem that the region is interfering?

**Comment by roelbarreto** [comment]
@danillocesar Can you execute this script on your firefox then post the log result here?
```javascript
function getFirstDayOfWeek() {
    return 1;
} 
var currentDate = new Date(),
                monthStartDate = new Date(currentDate.getFullYear(), currentDate.getMonth(), 1),
                monthEndDate = new Date(currentDate.getFullYear(), currentDate.getMonth() + 1, 0),
                calendarStartDate = new Date(currentDate.getFullYear(), currentDate.getMonth(), 1 - (monthStartDate.getDay() == 0 ? 6 : (monthStartDate.getDay() - getFirstDayOfWeek()))),
                calendarEndDate = new Date(currentDate.getFullYear(), currentDate.getMonth() + 1, 7 - (monthEndDate.getDay() - getFirstDayOfWeek())),
                calendar = calendarStartDate;

console.log(calendar, calendarEndDate);
```

**Comment by danillocesar** [comment]
Date 2017-08-28T03:00:00.000Z Date 2017-10-02T03:00:00.000Z


That's the result

**Comment by roelbarreto** [comment]
Got similar issue on Brazil timezone https://discussions.apple.com/thread/2715555?tstart=0
I'll continue investigating this
Thanks @danillocesar

**Comment by roelbarreto** [comment]
@danillocesar  A recent release related to your issue has been addressed, can you verify on demo page if still having the existing issue? https://ipiz.herokuapp.com/md-date-range-picker-demo/index.html

**Comment by danillocesar** [comment]
@ipiz  same problem, line is 334
` while (calendar < calendarEndDate) {
                dates.push(calendar);
                calendar = new Date(calendar.getFullYear(), calendar.getMonth(), calendar.getDate() + 1);
            }`

**Comment by roelbarreto** [comment]
@danillocesar  could you try another shot testing https://ipiz.herokuapp.com/md-date-range-picker-demo/index.html on your browser? I updated the demo to use the latest build i mate which limit the loop for possible infinite loop.

Regards...

**Comment by danillocesar** [comment]
@ipiz  yes, its not crashing now.. can i update my project with the new build?

**Comment by roelbarreto** [comment]
I can publish a new release with the fix 0.6.3

**Comment by roelbarreto** [comment]
You can now update your dependency version to 0.6.3 or get the latest build
Regards...
